aspnetcore/blazor/state-management.md

```diff
@@ -0,0 +1,357 @@
+
+`localStorage` and `sessionStorage` differ as follows:
+
+* `localStorage` is scoped to the user's browser. If the user reloads the page or closes and re-opens the browser, the state persists. If the user opens multiple browser tabs, the state is shared across the tabs. Data persists in `localStorage` until explicitly cleared.
+* `sessionStorage` is scoped to the user's browser tab. If the user reloads the tab, the state persists. If the user closes the tab or the browser, the state is lost. If the user opens multiple browser tabs, each tab has its own independent version of the data.
+
+Generally, using `sessionStorage` is safer. `sessionStorage` avoids the risk with `localStorage` that a user opens multiple tabs and encounters bugs or confusing behavior when a tab overwrites the state of other tabs. However, `localStorage` is the better choice if the app must persist state across closing and re-opening the browser.
+
+Caveats for using browser storage:
+
+* Similar to use of a server-side database, loading and saving are asynchronous.
+* Unlike a server-side database, storage isn't available during prerendering because the requested page doesn't exist in the browser during the prerendering stage.
+* Storage of a few kilobytes of data is reasonable to persist for Blazor server-side apps. Beyond a few kilobytes, you must consider the performance implications because the data is loaded and saved across the network.
+* Users may view or tamper with the data. Risks of tampering can be mitigated for Blazor server-side apps using [Data Protection](xref:security/data-protection/introduction).
+
+## Third-party browser storage solutions
+
+Third-party NuGet packages provide APIs for working with `localStorage` and `sessionStorage` in Blazor apps.
+
+It's worth considering choosing a package that transparently uses ASP.NET Core's [Data Protection](xref:security/data-protection/introduction) to encrypt the stored data and reduce the potential risk of tampering with stored data. If JSON-serialized data is stored in plaintext, users can not only see that data (for example, using browser developer tools) but also modify the stored data arbitrarily. Securing data isn't always a problem because the data might be trivial in nature (for example, setting the color of a UI element background). Avoid allowing users to inspect or tamper with sensitive data.
+
+## Protected Browser Storage experimental package
+
+An example of a NuGet package that provides [Data Protection](xref:security/data-protection/introduction) for `localStorage` and `sessionStorage` is [Microsoft.AspNetCore.ProtectedBrowserStorage](https://www.nuget.org/packages/Microsoft.AspNetCore.ProtectedBrowserStorage).
+
+> [!WARNING]
+> `Microsoft.AspNetCore.ProtectedBrowserStorage` is an unsupported experimental package unsuitable for production use at this time.
```

Comment:
Are we really saying `ProtectedBrowserStorage` is *unsuitable for production use*? It's been more carefully reviewed than probably any of the 3rd-party packages. I know it's unsupported, but that doesn't mean we want people not to use it.

cc @danroth27 @blowdart for comments on whether we need to advise people explicitly not to use it.

Reply:
My guidance thus far has been that the entire Blazor framework **_in preview_** isn't suitable (supported) for production workloads. Definitely *correct me* if that isn't the case.

Reply:
it's as unsuitable as the rest of blazor. So once blazor is deemed production ready, then I guess this is too.

Reply:
Just to be clear, `Microsoft.AspNetCore.ProtectedBrowserStorage` is not shipping as part of Blazor in 3.0, so it will remain experimental even after Blazor itself is RTMed.

Please do leave this warning here if that's the desired wording for all un-RTMed code.